src/libraries/System.Net.Quic/src/System/Net/Quic/QuicConnection.cs

```diff
@@ -609,6 +629,20 @@ private unsafe int HandleEventConnected(ref CONNECTED_DATA data)
     {
         _negotiatedApplicationProtocol = new SslApplicationProtocol(new Span<byte>(data.NegotiatedAlpn, data.NegotiatedAlpnLength).ToArray());
 
+        QUIC_HANDSHAKE_INFO info = MsQuicHelpers.GetMsQuicParameter<QUIC_HANDSHAKE_INFO>(_handle, QUIC_PARAM_TLS_HANDSHAKE_INFO);
+        _negotiatedCipherSuite = info.CipherSuite switch
+        {
+            QUIC_CIPHER_SUITE.TLS_AES_128_GCM_SHA256 => TlsCipherSuite.TLS_AES_128_GCM_SHA256,
+            QUIC_CIPHER_SUITE.TLS_AES_256_GCM_SHA384 => TlsCipherSuite.TLS_AES_256_GCM_SHA384,
+            QUIC_CIPHER_SUITE.TLS_CHACHA20_POLY1305_SHA256 => TlsCipherSuite.TLS_CHACHA20_POLY1305_SHA256,
+            _ => default
+        };
+        _negotiatedSslProtocol = info.TlsProtocolVersion switch
```

Comment:
I'm wondering if we need to bother at all ... at least until there is Tls 1.4. If anythiung I would Debug.Assert here as reminder.

Reply:
we actually don't need to bother with either of the switches, the enum values are the same so we can cast directly.

I added assert for TLS 1.3 and unit test for cipher suites so that we notice if something changes.